### PR TITLE
E2E tests with cypress

### PR DESCRIPTION
**Pull request by OmriBarZik**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/pojome/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe: e2e tests

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
+    mysql:
+    wordpress_e2e_tests:
+    cli_e2e_tests:
+
 workflows:
   version: 2
+
+  e2e:
+    triggers:
+      - schedule:
```
Comment (KingYes) [inline .circleci/config.yml]: fix indent

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,29 +1,51 @@
...
+  working_directory: ~/elementor
+  machine:
+    mysql:
+    wordpress_e2e_tests:
+    cli_e2e_tests:
+
 workflows:
   version: 2
+
+  e2e:
```
Comment (KingYes) [inline .circleci/config.yml]: Run only in PRs

File: .circleci/config.yml [hunk 3]
```diff
@@ -49,182 +71,237 @@ job-references:
...
+          name: Prepare Elementor
+          command: |
+            mv cypress.json build/cypress.json
+            mv docker-compose.yml build/docker-compose.yml
+            mv tests build/tests
+            mv bin build/bin
+            mv package.json build/package.json
+            mv build ../build
+            cd ..
+            rm -fr elementor
```
Comment (KingYes) [inline .circleci/config.yml]: -rf

File: .circleci/config.yml [hunk 4]
```diff
@@ -49,182 +71,237 @@ job-references:
...
+          paths:
+            - elementor
+
+  e2e_electron:
+    <<: *e2e_default
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Setup Wordpress
```
Comment (KingYes) [inline .circleci/config.yml]: capelcase

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,29 +1,51 @@
+defaults: &e2e_default
```
Comment (KingYes) [inline .circleci/config.yml]: try multiple files

File: bin/install-wordpress.sh [hunk 6]
```diff
@@ -52,6 +52,7 @@ echo -e $(status_message "Installing WordPress...")
 # The `-u 33` flag tells Docker to run the command as a particular user and
 # prevents permissions errors. See: https://github.com/WordPress/gutenberg/pull/8427#issuecomment-410232369
 docker-compose $DOCKER_COMPOSE_FILE_OPTIONS run --rm -u 33 $CLI core install --title="$SITE_TITLE" --admin_user=admin --admin_password=password --admin_email=admin@localhost.local --skip-email --url=http://localhost:$HOST_PORT >/dev/null
+docker-compose $DOCKER_COMPOSE_FILE_OPTIONS run --rm -u 33 $CLI config set AUTOSAVE_INTERVAL 2 --add --type=constant >/dev/null
```
Comment (KingYes) [inline bin/install-wordpress.sh]: try as filter

File: cypress.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  "baseUrl": "http://localhost:4001",
+  "pageLoadTimeout": 10000,
+  "integrationFolder": "tests/e2e/integration",
+  "supportFile": "tests/e2e/support/index.js",
+  "pluginsFile": "tests/e2e/plugins/index.js",
+  "fixturesFolder": "tests/e2e/fixtures",
+  "fileServerFolder": "log/application-data",
+  "screenshotsFolder": "log/screenshots",
+  "videosFolder": "log/videos",
+  "projectId": "uta8ca"
```
Comment (KingYes) [inline cypress.json]: remove

File: package.json [hunk 8]
```diff
@@ -1,53 +1,53 @@
...
-		"grunt-webpack": "^3.1.2",
-		"grunt-wp-readme-to-markdown-with-extra": "~2.2.0",
-		"load-grunt-tasks": "^4.0.0",
-		"path": "^0.12.7",
-		"selenium-server": "^3.14.0",
-		"uglifyjs-webpack-plugin": "^1.3.0",
-		"webpack": "^4.16.5",
-		"nightwatch": "^0.9.21"
-	}
+  "name": "elementor",
```
Comment (KingYes) [inline package.json]: tabs plz!

File: tests/e2e/integration/auto-save.spec.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
+/// <reference types="Cypress" />
+
+var setPanelSelectedElement = function( elementor, category, name ) {
```
Comment (KingYes) [inline tests/e2e/integration/auto-save.spec.js]: es6

**Review by KingYes — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 5 above)
Comment (OmriBarZik) [reply .circleci/config.yml]: there is no solution

**Pull request merged**